File: algorithms/204. Count Primes/solution.py
class Solution(object):
    def countPrimes(self, n):
        """
        :type n: int
        :rtype: int
        """
        if n < 2:
            return 0
        isPrime = [1] * n
        for i in xrange(2, int(n ** 0.5) + 1):
            if isPrime[i]:
                for j in xrange(i * i, n, i):
                    isPrime[j] = 0
        return sum(isPrime) - 2

# 20 / 20 test cases passed.
# Status: Accepted
# Runtime: 795 ms
# beats 72.09%

File: algorithms/204. Count Primes/solution.cc
class Solution {
public:
    int countPrimes(int n) {
        if ( n < 2 ) {
            return 0;
        }
        vector<bool> isPrime(n, true);
        isPrime[0] = isPrime[1] = false;
        int sqrtn = int(sqrt(n));
        for ( int i = 2; i <= sqrtn; ++i ) {
            if ( isPrime[i] ) {
                for ( int j = i * i; j < n; j += i ) {
                    isPrime[j] = false;
                }
            }
        }
        return count(isPrime.begin(), isPrime.end(), true);
    }
};

// 20 / 20 test cases passed.
// Status: Accepted
// Runtime: 46 ms
// beats 43.82%
